### PR TITLE
Add album detail view

### PR DESCRIPTION
**Pull request by danbee**
Screenshot 2019-06-07 at 11 47 28

### CODE REVIEW

File: Persephone/Controllers/AlbumDetailView.swift [hunk 1]
```diff
@@ -47,16 +47,32 @@ class AlbumDetailView: NSViewController {
...
   func getAlbumSongs(for album: Album) {
     App.mpdClient.getAlbumSongs(for: album.mpdAlbum) { (mpdSongs: [MPDClient.MPDSong]) in
-      self.dataSource.albumTracks = mpdSongs.map {
-        return Song(mpdSong: $0)
-      }
+      self.dataSource.setAlbumSongs(
+        mpdSongs.map { Song(mpdSong: $0) }
+      )
+
+      self.getBigCoverArt(song: self.dataSource.albumSongs.first!.song ?? self.dataSource.albumSongs[1].song!)
```
Comment (sharplet) [inline Persephone/Controllers/AlbumDetailView.swift]: What about something like:

```swift
self.dataSource.albumSongs.compactMap { $0.song }.first
```

File: Persephone/Controllers/AlbumDetailView.swift [hunk 2]
```diff
@@ -47,16 +47,32 @@ class AlbumDetailView: NSViewController {
...
         self.albumTracksView.reloadData()
       }
     }
   }
 
+  func getBigCoverArt(song: Song) {
+    let coverArtService = CoverArtService(song: song)
+
+    coverArtService.fetchBigCoverArt()
+      .done() { image in
```
Comment (sharplet) [inline Persephone/Controllers/AlbumDetailView.swift]: Does `done()` have an optional parameter to switch queues?

File: Persephone/Controllers/AlbumDetailView.swift [hunk 3]
```diff
@@ -46,15 +46,15 @@ class AlbumDetailView: NSViewController {
   }
 
   func getAlbumSongs(for album: Album) {
-    App.mpdClient.getAlbumSongs(for: album.mpdAlbum) { (mpdSongs: [MPDClient.MPDSong]) in
-      self.dataSource.setAlbumSongs(
+    App.mpdClient.getAlbumSongs(for: album.mpdAlbum) { [weak self] (mpdSongs: [MPDClient.MPDSong]) in
+      self?.dataSource.setAlbumSongs(
```
Comment (sharplet) [inline Persephone/Controllers/AlbumDetailView.swift]: Maybe `guard let self = self` to ensure that the reference remains valid for the duration of the block?

File: Persephone/Controllers/AlbumDetailView.swift (hunk 2 above)
Comment (danbee) [reply Persephone/Controllers/AlbumDetailView.swift]: It does!

File: Persephone/Controllers/AlbumDetailView.swift (hunk 1 above)
Comment (danbee) [reply Persephone/Controllers/AlbumDetailView.swift]: I completely refactored this bit. I intended to before making the PR but forgot all about it.

**Pull request merged**